test/run_notebooks.sh

```diff
@@ -4,13 +4,23 @@ RED='\033[0;31m'
 GREEN='\033[0;32m'
 NC='\033[0m' # No Color
 
-## Returns the correct image name.  If the test name's
-## environmental variable exists, return that, otherwise return the default
-## value for the image.
+SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+
+
+## Returns the correct image name to run a notebook.
+## If the test name's environmental variable exists, return that.
+## otherwise:
+##   if the second argument passed is 'true' return the tag value for the image read from the version.txt file.
+##   else returns 'latest'
```

Comment:
Something that could be ignored, but...

This function and the `image_name` in `test-in-bamboo.sh` are essentially the same (except the latter just hard-codes the `$use_versions == True` branch). Maybe it's overkill, but having this function in one place, and then using `source ./that/one/place.sh` would reduce duplication across the two scripts.

Reply:
(This potentially feels like the tip of the iceberg for trying to deduplicate stuff and streamline some of these scripts - I know that we've both had trouble tracking the path through the scripts at times)

Reply:
this is the tip of the iceberg, but of me not knowing bash.  I suppose that's not a hard fix. It's a bit awkward with the locations of the files, but that probably points to other issues.  I'll see if I can clean it up.

Reply:
Sure - if it looks like a big ask, then we can specifically write a ticket to make those changes. The long IP is on the horizon... (and quickly filling up)

Reply:
I don't love it.  but [6112147](https://github.com/nasa/harmony-regression-tests/pull/74/commits/611214736f97908e1c54fc39ee5fe2a7dbdd563e)